src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -338,17 +338,34 @@ private async Task RecoverStoreTipAsync(ChainedHeader chainedHeader)
                 throw new ProvenBlockHeaderException("Proven block header store failed to recover.");
             }
 
-            if (tipHeight > chainedHeader.Height)
+            ChainedHeader pendingTip = null;
```

Comment:
`ChainedHeader pendingTip = null`  can go inside the if statement `if (newChainedHeader.Height <= tipHeight)`
And also the code bellow `  if (pendingTip != null)` can go in the end of the if statement.

Reply:
Ok.